modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/search/index/InstanceWorkflowMetricsIndexerImpl.java

```diff
@@ -146,66 +298,56 @@ public void reindex(long companyId) throws PortalException {
 				dynamicQuery.add(companyIdProperty.eq(companyId));
 			});
 		actionableDynamicQuery.setPerformActionMethod(
-			(KaleoInstance kaleoInstance) ->
-				workflowMetricsPortalExecutor.execute(
-					() -> addDocument(createDocument(kaleoInstance))));
+			(KaleoInstance kaleoInstance) -> {
+				KaleoDefinitionVersion kaleoDefinitionVersion =
+					getKaleoDefinitionVersion(
+						kaleoInstance.getKaleoDefinitionVersionId());
+
+				if (Objects.isNull(kaleoDefinitionVersion)) {
+					return;
+				}
+
+				long duration = 0;
```

Comment:
duration = 0 ?

duration = Duration.between(createDate.toInstant(), completionDate.toInstant()).toMillis();?

Reply:
better, remove the duration, we just need the dates, we can do this calculation as you did. `_getDuration`